R/fit_model.R

```diff
@@ -0,0 +1,179 @@
+  )
+  truncation <- format_right_truncation(
+    parameters[["right_truncation"]],
+    data
+  )
+
+  # Stan sampler ------------------------------------------------------------
+  stan <- EpiNow2::stan_opts(
+    cores = sampler_opts[["cores"]],
+    chains = sampler_opts[["chains"]],
+    # NOTE: seed gets used twice -- as the seed passed here to the Stan sampler
+    # and below as the R PRNG seed for EpiNow2 initialization
+    seed = seed,
+    warmup = sampler_opts[["iter_warmup"]],
+    samples = sampler_opts[["iter_samples"]],
+    control = list(
+      adapt_delta = sampler_opts[["adapt_delta"]],
+      max_treedepth = sampler_opts[["max_treedepth"]]
+    )
+  )
+
+  df <- data.frame(
+    confirm = data[["confirm"]],
+    date = as.Date(data[["reference_date"]])
+  )
+  rlang::try_fetch(
+    withr::with_seed(seed, {
+      EpiNow2::epinow(
+        df,
+        generation_time = generation_time,
+        delays = delays,
+        truncation = truncation,
+        horizon = horizon,
+        rt = rt,
+        gp = gp,
+        stan = stan,
+        verbose = interactive()
+      )
+    }),
+    # Downgrade model erroring out to a warning so we can catch and return
+    error = function(cnd) {
+      cli::cli_warn(
+        "Model fitting failed. Returning NA.",
+        parent = cnd,
+        class = "failing_fit"
+      )
+      NA
+    }
+  )
+}
+
+#' Format PMFs for EpiNow2
+#'
+#' Opinionated wrappers around EpiNow2::generation_time_opts(),
+#' EpiNow2::delay_opts(), or EpiNow2::dist_spec() that
+#' formats the generation interval, delay, or right truncation parameters
+#' as an object ready for input to EpiNow2.
+#'
+#' Delays or right truncation are optional and
+#' can be skipped by passing `pmf = NA`.
+#'
+#' @param pmf As returned from [CFAEpiNow2Pipeline::read_disease_parameters()].
+#'   A PMF vector or an NA, if not applying the PMF to the model fit.
+#'
+#' @return An EpiNow2::*_opts() formatted object or NA with a message
+#' @name opts_formatter
+NULL
+
+#' @rdname opts_formatter
+#' @export
+format_generation_interval <- function(pmf) {
+  if (
+    rlang::is_na(pmf) || rlang::is_null(pmf)
+  ) {
+    cli::cli_abort("No generation time PMF specified but is required",
+      class = "Missing_GI"
+    )
+  }
+
+  suppressWarnings({
+    EpiNow2::generation_time_opts(
+      dist = EpiNow2::dist_spec(
+        pmf = pmf
+      )
+    )
+  })
+}
+
+#' @rdname opts_formatter
+#' @export
+format_delay_interval <- function(pmf) {
+  if (rlang::is_na(pmf) || rlang::is_null(pmf)) {
+    cli::cli_alert("Not adjusting for infection to case delay")
+    EpiNow2::delay_opts()
+  } else {
+    suppressWarnings({
+      EpiNow2::delay_opts(
+        dist = EpiNow2::dist_spec(
+          pmf = pmf
+        )
+      )
+    })
+  }
+}
+
+#' @inheritParams fit_model
+#' @rdname opts_formatter
+#' @export
+format_right_truncation <- function(pmf, data) {
+  if (
+    rlang::is_na(pmf) || rlang::is_null(pmf)
```

Comment:
This one (and the first one) are on new lines but the middle one isn't. They should be the same length so seems like style can be standardised

Reply:
https://github.com/CDCgov/cfa-epinow2-pipeline/pull/26/commits/e849eb52a9c8726cfff35efef13be60df9fe1c2a